core/src/franchise/player/player.controller.ts

```diff
@@ -154,4 +166,77 @@ export class PlayerController {
 
         return "SUCCESS";
     }
+
+    @MessagePattern(CoreEndpoint.GetPlayerByPlatformId)
+    async getPlayerByPlatformId(@Payload() payload: unknown): Promise<CoreOutput<CoreEndpoint.GetPlayerByPlatformId>> {
+        const data = CoreSchemas[CoreEndpoint.GetPlayerByPlatformId].input.parse(payload);
+
+        const game = await this.gameService.getGameById(data.gameId);
+        const platform = await this.platformService.getPlatformByCode(data.platform);
+        const player = await this.playerService.getPlayerByGameAndPlatform(game.id, platform.id, data.platformId).catch(() => null);
+
+        if (!player) return {
+            success: false,
+            request: data,
+        };
```

Comment:
Rather than `.catch(() => null)`, then checking for null, just put the `getPlayerByGameAndPlatform` call into a `try`/`catch`

We should also be looking for a specific error (player not found), because if its a different unexpected error we should probably halt execution. This may take some more refactoring though and not be doable in this PR.